src/lib.rs

```diff
@@ -1066,12 +1058,26 @@ impl<A: Array> SmallVec<A> {
                 );
             }
 
+            // There are no more duplicate or uninitialized slots, so the guard is not needed.
             self.set_len(old_len + num_added);
+            mem::forget(guard);
+        }
+
+        unsafe {
+            for element in iter {
```

Comment:
Can this be just:

```
for element in iter {
    self.insert(index + num_added, element);
    num_added += 1;
}
```

And remove the unsafe block?

Reply:
Great idea.  Done.